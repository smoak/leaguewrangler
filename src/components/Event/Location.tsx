import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';
import React, { FC } from 'react';

interface LocationProps {
  readonly locationName: string;
  readonly locationUrl?: string;
}

export const Location: FC<LocationProps> = ({ locationName, locationUrl }) => {
  if (!locationUrl) {
    <Typography component="p">{locationName}</Typography>;
  }

  return (
    <Typography component="p">
      <Link href={locationUrl} target="_blank" rel="noreferrer">
        {locationName}
      </Link>
    </Typography>
  );
};
