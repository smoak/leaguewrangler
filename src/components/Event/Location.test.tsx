import { ShallowWrapper, shallow } from 'enzyme';
import React from 'react';

import { Location } from './Location';

describe('Location', () => {
  let component: ShallowWrapper;

  describe('with no locationUrl', () => {
    beforeEach(() => {
      component = shallow(<Location locationName="locationName" />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with a locationUrl', () => {
    beforeEach(() => {
      component = shallow(<Location locationName="locationName" locationUrl="https://example.com" />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
