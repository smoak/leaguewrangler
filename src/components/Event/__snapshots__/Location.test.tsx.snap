// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location with a locationUrl renders as expected 1`] = `
<WithStyles(ForwardRef(Typography))
  component="p"
>
  <WithStyles(ForwardRef(Link))
    href="https://example.com"
    rel="noreferrer"
    target="_blank"
  >
    locationName
  </WithStyles(ForwardRef(Link))>
</WithStyles(ForwardRef(Typography))>
`;

exports[`Location with no locationUrl renders as expected 1`] = `
<WithStyles(ForwardRef(Typography))
  component="p"
>
  <WithStyles(ForwardRef(Link))
    rel="noreferrer"
    target="_blank"
  >
    locationName
  </WithStyles(ForwardRef(Link))>
</WithStyles(ForwardRef(Typography))>
`;
